component/silence.jsonnet

```diff
@@ -0,0 +1,105 @@
+local kap = import 'lib/kapitan.libjsonnet';
+local kube = import 'lib/kube.libjsonnet';
+
+local inv = kap.inventory();
+local params = inv.parameters.openshift4_monitoring;
+
+local namespace = {
+  metadata+: {
+    namespace: params.namespace,
+  },
+};
+
+local cm = kube.ConfigMap('silence') + namespace {
+  data: {
+    silence: |||
+      #!/bin/bash
+
+      curl_opts=( https://alertmanager-main.openshift-monitoring.svc.cluster.local:9095/api/v2/silences --cacert /etc/ssl/certs/serving-certs/service-ca.crt --header 'Content-Type: application/json' --header "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" --resolve alertmanager-main.openshift-monitoring.svc.cluster.local:9095:$(getent hosts alertmanager-operated.openshift-monitoring.svc.cluster.local | awk '{print $1}' | head -n 1) --silent )
```

Comment:
What's wrong with using `alertmanager-main` directly?

Reply:
The certificate name does not match.

Reply:
But apparently it matches `alertmanager-main.openshift-monitoring.svc.cluster.local`, so why not just use this (without the `--resolve`)?

Reply:
For some reason, I was unable to connect to port 9095 on that service.